native/explorer/src/series.rs

```diff
@@ -548,7 +548,52 @@ pub fn s_in(s: ExSeries, rhs: ExSeries) -> Result<ExSeries, ExplorerError> {
         DataType::Date => s.date()?.is_in(&rhs)?,
         DataType::Time => s.time()?.is_in(&rhs)?,
         DataType::Datetime(_, _) => s.datetime()?.is_in(&rhs)?,
-        dt => panic!("is_in/2 not implemented for {dt:?}"),
+        DataType::Categorical(Some(mapping)) => {
+            let l_logical = s.categorical()?.logical();
+
+            match rhs.dtype() {
+                DataType::Utf8 => {
+                    let mut r_ids: Vec<Option<u32>> = vec![];
+
+                    // In case the right-hand is a series of strings, we only care
+                    // about members in the category on the left, or if it's None.
+                    for opt in rhs.unique()?.utf8()?.into_iter() {
+                        match opt {
+                            Some(slice) => {
+                                if let Some(id) = mapping.find(slice) {
+                                    r_ids.push(Some(id));
+                                }
+                            }
+                            None => r_ids.push(None),
+                        }
+                    }
+
+                    let r_logical = Series::new("r_logical", r_ids);
+
+                    l_logical.is_in(&r_logical)?
+                }
+                DataType::Categorical(Some(_rhs_mapping)) => {
+                    let mut r_ids: Vec<Option<u32>> = vec![];
+                    for opt in rhs.categorical()?.iter_str() {
+                        match opt {
+                            Some(slice) => {
+                                if let Some(id) = mapping.find(slice) {
+                                    r_ids.push(Some(id));
+                                }
+                            }
+                            None => r_ids.push(None),
+                        }
+                    }
```

Comment:
So the idea is that we don't need this branch either, both categories should point to the same mapping in memory by default (that's why Polars raise and that's why we changed categorize to support strings, so we can build a category from an existing dtype=category sharing the same mapping).

Reply:
Yeah, I know we discussed this previously. The problem was only with the cases where `categorise` a string series would create a series with `nils` for elements that are not present. We would  have a new category series that would return `true` for `nils`. I could remove the nil elements, but I wouldn't know if the original series contained a `nil`.

## Examples

```elixir
cats = Explorer.Series.from_list(["a", "b", "c", nil], dtype: :category)
strings = Explorer.Series.from_list(["b", "a", "d", "c"])

new_cats = Explorer.Series.categorise(strings, cats)
#=> #Explorer.Series<
#=>  Polars[4]
#=>  category ["b", "a", nil, "c"]
#=> >
```

And if we use `Series.in/2`:

```elixir
Explorer.Series.in(cats, new_cats)
#=> #Explorer.Series<
#=>   Polars[4]
#=>   boolean [true, true, true, true]
#=> >
```
Since the `nil` was introduced in the `new_cats` because `d` was not an element, it is considering `nil` as an element.
But I think this is not what the user is expecting.

I guess we could require that the comparison - in `Series.in/2` - would work only with the same `:category` dtype, and we could assume that the person did the categorisation and the filter of nils correctly. WDYT?

Reply:
I think that’s true for all series though, no? If I have a list of integers that are nils, and I check against a series of integers with nil, then they will be all true too, right?

Reply:
Yes, but that is expected. I think if we use `categorise/2` underneath - in the implementation of `Series.in/2`, and the categorised series contains a `nil` that was a result of a missing element, this is going to be a bug. Makes sense?